homeassistant/components/mqtt/lock.py

```diff
@@ -101,13 +103,19 @@ def __init__(self, config, config_entry, discovery_hash):
     async def discovery_update(self, discovery_payload):
         """Handle updated discovery message."""
         config = PLATFORM_SCHEMA(discovery_payload)
-        self._config = config
+        self._setup_from_config(config)
         await self.attributes_discovery_update(config)
         await self.availability_discovery_update(config)
         await self.device_info_discovery_update(config)
         await self._subscribe_topics()
         self.async_write_ha_state()
 
+    def _setup_from_config(self, config):
+        """(Re)Setup the entity."""
+        self._config = config
+
+        self._optimistic = config[CONF_OPTIMISTIC]
```

Comment:
Shouldn't we make this a `@property` instead? Then the existing code works fine and we don't need to worry about potentially having the `_config` being out of sync with its values (like `optimistic`)

Reply:
The same concept is used in all MQTT platforms. The suggestion makes sense, but I'd prefer to leave that for another PR.

Reply:
Single source of truth for the win. Let's make it a property, ok to wait for a future PR.